pkg/builder/builder.go

```diff
@@ -474,6 +476,22 @@ func (b Builder) CheckValidation() error {
 			if len(stack.MixedInstancesPolicy.Override) == 0 {
 				return errors.New("you have to set at least one instance type to use in override")
 			}
+
+			//Check Input override spot instance type
+			spotInstanceTypes := strings.Split(b.Config.OverrideSpotInstanceType, "|")
+			var intelTypeCount, armTypeCount = 0, 0
+			for _, spotInstanceType := range spotInstanceTypes {
+				instanceTypeCategory := strings.Split(spotInstanceType, ".")
+				if Contains(armTypeList, instanceTypeCategory[0]) {
+					armTypeCount++
+				} else {
+					intelTypeCount++
+				}
+			}
+			spotInstanceTypeCount := len(spotInstanceTypes)
+			if !((intelTypeCount == spotInstanceTypeCount) || (armTypeCount == spotInstanceTypeCount)) {
+				return errors.New("you can only use same type of spot instance type(arm64 and intel_x86 type)")
```

Comment:
You have to add test cases in `builder_test.go` for these cases.

Reply:
Does this only work with `OverrideSpotInstanceType` ?
What if it is not defined or passed with different delimiter (like t2.nano,t3nano...)?

And, I think only counting arm64 instance type is enough to validate.

Reply:
Thank you to your advice. I think it must fix logic, too. so I add another delimiter cases.